lib/dialects/snowflake/query-generator.js

```diff
@@ -0,0 +1,633 @@
+        context: 'addColumn',
+        tableName: table,
+        foreignKey: key
+      }),
+      ';'
+    ]);
+  }
+
+  removeColumnQuery(tableName, attributeName) {
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'DROP',
+      this.quoteIdentifier(attributeName),
+      ';'
+    ]);
+  }
+
+  changeColumnQuery(tableName, attributes) {
+    const query = subQuery => `ALTER TABLE ${this.quoteTable(tableName)} ALTER COLUMN ${subQuery};`;
+    const sql = [];
+    for (const attributeName in attributes) {
+      let definition = this.dataTypeMapping(tableName, attributeName, attributes[attributeName]);
+      let attrSql = '';
+
+      if (definition.includes('NOT NULL')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} SET NOT NULL`);
+
+        definition = definition.replace('NOT NULL', '').trim();
+      } else if (!definition.includes('REFERENCES')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} DROP NOT NULL`);
+      }
+
+      if (definition.includes('DEFAULT')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} SET DEFAULT ${definition.match(/DEFAULT ([^;]+)/)[1]}`);
+
+        definition = definition.replace(/(DEFAULT[^;]+)/, '').trim();
+      } else if (!definition.includes('REFERENCES')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} DROP DEFAULT`);
+      }
+
+      if (attributes[attributeName].startsWith('ENUM(')) {
+        attrSql += this.pgEnum(tableName, attributeName, attributes[attributeName]);
+        definition = definition.replace(/^ENUM\(.+\)/, this.pgEnumName(tableName, attributeName, { schema: false }));
+        definition += ` USING (${this.quoteIdentifier(attributeName)}::${this.pgEnumName(tableName, attributeName)})`;
+      }
+
+      if (definition.match(/UNIQUE;*$/)) {
+        definition = definition.replace(/UNIQUE;*$/, '');
+        attrSql += query(`ADD UNIQUE (${this.quoteIdentifier(attributeName)})`).replace('ALTER COLUMN', '');
+      }
+
+      if (definition.includes('REFERENCES')) {
+        definition = definition.replace(/.+?(?=REFERENCES)/, '');
+        attrSql += query(`ADD FOREIGN KEY (${this.quoteIdentifier(attributeName)}) ${definition}`).replace('ALTER COLUMN', '');
+      } else {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} TYPE ${definition}`);
+      }
+
+      sql.push(attrSql);
+    }
+
+    return sql.join('');
+  }
+
+  renameColumnQuery(tableName, attrBefore, attributes) {
+    const attrString = [];
+
+    for (const attrName in attributes) {
+      const definition = attributes[attrName];
+      attrString.push(`'${attrBefore}' '${attrName}' ${definition}`);
+    }
+
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'RENAME COLUMN',
+      attrString.join(' to '),
+      ';'
+    ]);
+  }
+
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
```

Comment:
I have no idea what `smth` is - can you rename and/or just explain?

Reply:
This is used acorss the code base...I will rename it to `attr` in snowflake branch